test/cli/credentials/list.test.js

```diff
@@ -44,58 +46,60 @@ describe('eg credentials list -c ', () => {
     return credentials
       .create(username, type, options)
       .then((credential) => {
-        createdTypes.inc(type, isActive);
+        createdTypes.inc(credential, type, isActive);
 
         if (type === 'key-auth') {
-          const { keyId } = credential;
-          createdKeyAuthKeys.add(keyId, isActive);
+          createdKeyAuthKeys.add(credential.id, isActive);
+        }
 
-          if (!isActive) {
-            return credentials.deactivate(keyId, type);
-          }
+        if (!isActive) {
+          return credentials.deactivate(credential.id, type);
         }
       });
   };
 
-  before(() => adminHelper.start());
-  after(() => adminHelper.stop());
-
-  beforeEach(() => {
-    ({ program, env } = environment.bootstrap());
+  before(() => {
     createdTypes.reset();
     createdKeyAuthKeys.reset();
 
-    env.prepareHijack();
-
     return adminHelper
-      .reset()
-      .then(() => {
-        return adminHelper
-          .admin
-          .users
-          .create({
-            username: idGen.v4(),
-            firstname: 'La',
-            lastname: 'Deeda'
-          })
-          .then(user => {
-            username = user.username;
-            return Promise.all([
-              createCredential(user.username, 'key-auth'),
-              createCredential(user.username, 'basic-auth', { password: 'test1' }),
-              createCredential(user.username, 'oauth2', { secret: 'eg1' }),
-              createCredential(user.username, 'key-auth', {}, false),
-              createCredential(user.username, 'key-auth', {}, false),
-              createCredential(user.username, 'key-auth', {}, false)
-            ]);
-          });
+      .start()
+      .then(() => adminHelper.admin.users.create({
+        username: idGen.v4(),
+        firstname: 'La',
+        lastname: 'Deeda'
+      }))
+      .then(user => {
+        username = user.username;
+        return Promise.all([
+          createCredential(user.username, 'key-auth'),
+          createCredential(user.username, 'basic-auth', { password: 'test1' }),
+          createCredential(user.username, 'oauth2', { secret: 'eg1' }),
+          createCredential(user.username, 'key-auth', {}, false),
+          createCredential(user.username, 'key-auth', {}, false),
+          createCredential(user.username, 'key-auth', {}, false)
+        ]);
       });
   });
 
+  after(() => adminHelper.stop());
+
+  beforeEach(() => {
+    ({ program, env } = environment.bootstrap());
+    env.prepareHijack();
+  });
+
   afterEach(() => {
     env.resetHijack();
   });
 
+  it('should have the id property, but not the secret ones', () => {
+    createdTypes.raw.forEach(cred => {
+      should(cred).not.have.properties('secret', 'password');
```

Comment:
If you remove them from response how would I know autogenerated values for a password? Insert op should still have a secret in response

Reply:
This is not an insert operation, it's a listing one. See the [ran command](https://github.com/ExpressGateway/express-gateway/pull/687/files#diff-c7a7013b987baf8de10bb621bf51006eR160)